traitlets/traitlets.py

```diff
@@ -801,6 +800,12 @@ def class_own_traits(cls, **metadata):
         return {n: t for (n, t) in cls.class_traits(**metadata).items()
                 if getattr(sup, n, None) is not t}
 
+    def has_trait(self, name):
+        """Returns True if the specified name is the name of a trait
+        attribute."""
+        return (hasattr(self.__class__, name) and
+                isinstance(getattr(self.__class__, name), TraitType))
```

Comment:
getattr takes an optional default parameter, so we should be able to save the separate `hasattr` check and just do `isinstance(getattr(self.__class__, name, None), TraitType)`.